memestra/caching.py

```diff
@@ -2,12 +2,66 @@
 import hashlib
 import yaml
 
```

Comment:
I used to have an issue with caching that I can't reproduce anymore.
Very cool :)